Mathlib/Order/Hom/WithTopBot.lean

```diff
@@ -0,0 +1,242 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Order.Hom.BoundedLattice
+
+/-!
+# Adjoining `⊤` and `⊥` to lattice homomorphisms
+
+This file defines ways to adjoin `⊤` or `⊥` or both to lattice homomorphisms, and properties about
+the results. Some definitions cause a possibly unbounded homomorphism to become bounded, so they
+change the type of the homomorphism.
```

Comment:
Did you try also moving definitions like `OrderHom.withTopMap` here? The name of this file suggests that it should not only be restricted to lattices.

Reply:
I've moved everything `WithTopBot`-related to `Order.Hom.WithTopBot`.